lib/streamlit/elements/image.py

```diff
@@ -488,8 +538,17 @@ def marshall_images(
                     image = textfile.read()
 
             # Following regex allows svg image files to start either via a "<?xml...>" tag eventually followed by a "<svg...>" tag or directly starting with a "<svg>" tag
-            if re.search(r"(^\s?(<\?xml[\s\S]*<svg\s)|^\s?<svg\s)", image):
-                proto_img.markup = f"data:image/svg+xml,{image}"
+            if re.search(r"(^\s?(<\?xml[\s\S]*<svg\s)|^\s?<svg\s|^\s?<svg>\s)", image):
+                if _check_if_svg_can_be_rendered_as_img(image):
+                    try:
+                        proto_img.url = f"data:image/svg+xml,{_normalize_svg(image)}"
+                    except Exception as e:
+                        warnings.warn(
```

Comment:
I don't think we are currently using the `warnings` module anywhere else in the codebase. Maybe it is better to use our logger, for example:

https://github.com/streamlit/streamlit/blob/0dfd319409229d80667b370c664d87f75812ca2d/lib/streamlit/__init__.py#L489